app/services/scheduler/scheduler.go

```diff
@@ -76,18 +89,20 @@ func (s *Scheduler) Schedule(id string, submission ethsubmission.Submission) err
 }
 
 // Cancel - Removes and cancels an already scheduled Transaction
-func (s *Scheduler) Cancel(id string) error {
+func (s *Scheduler) Cancel(id string, submitterAddress string) error {
 	t, exists := s.tasks.Load(id)
 	if !exists {
 		s.logger.Warnf("Scheduled transaction execution for [%s] not found.", id)
 		return nil
 	}
-	s.tasks.Delete(id)
 
-	timer := t.(*time.Timer)
-	timer.Stop()
+	storage := t.(*scheduler.Storage)
+	storage.Timer.Stop()
```

Comment:
I think that stopping the timer should be inside the `if`. We should be stopping the timer only if the submitter is != current operator address